extension.js

```diff
@@ -354,6 +360,17 @@ function update_height_setting(win) {
     const workarea = workarea_for_window(win);
     const current_height = win.get_frame_rect().height / workarea.height;
     settings.set_double('window-height', Math.min(1.0, current_height));
+    settings.set_boolean('window-maximize', false);
```

Comment:
`update_height_setting` is called when resize completes (i. e. when you release the mouse button). I think it's not what you want here. It may be called in response to some other events too, but please don't rely on that.

Reply:
For example, it's not called (at least, should not be) when you change window height from Preferences dialog